File: micro-common/src/main/java/com/jiangtj/micro/common/validation/MobilePhoneStringValidator.java
package com.jiangtj.micro.common.validation;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

import java.util.regex.Pattern;

public class MobilePhoneStringValidator implements ConstraintValidator<MobilePhone, String> {

    static Pattern p = Pattern.compile("^1[0-9]{10}$");

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        if (value == null) {
            return true;
        }
        return p.matcher(value).matches();
    }
}

File: micro-common/src/main/java/com/jiangtj/micro/common/validation/MobilePhoneStringValidator.kt
package com.jiangtj.micro.common.validation

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import java.util.regex.Pattern

class MobilePhoneStringValidator : ConstraintValidator<MobilePhone, String?> {
    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
        if (value == null) {
            return true
        }
        return p.matcher(value).matches()
    }

    companion object {
        var p: Pattern = Pattern.compile("^1[0-9]{10}$")
    }
}
